README.md

```diff
@@ -2,727 +2,369 @@
-lrwxrwxrwx root   9 Feb 19 12:35 usb-MBED_microcontroller_066EFF525257775087141721-0:0 -> ../../sdd
-lrwxrwxrwx root   9 Feb 19 12:35 usb-MBED_microcontroller_066EFF534951775087215736-0:0 -> ../../sdc
-lrwxrwxrwx root   9 Dec  3 16:10 usb-MBED_microcontroller_0670FF494956805087154420-0:0 -> ../../sdc
-lrwxrwxrwx root   9 Feb 19 12:35 usb-mbed_Microcontroller_101000000000000000000002F7F18695-0:0 -> ../../sde
-lrwxrwxrwx root   9 Dec  3 09:10 wwn-0x5000cca30ccffb77 -> ../../sda
-lrwxrwxrwx root  10 Dec  3 09:10 wwn-0x5000cca30ccffb77-part1 -> ../../sda1
-lrwxrwxrwx root  10 Dec  3 09:10 wwn-0x5000cca30ccffb77-part2 -> ../../sda2
-lrwxrwxrwx root  10 Dec  3 09:10 wwn-0x5000cca30ccffb77-part5 -> ../../sda5
+```python
+>>> import mbed_lstools
+>>> mbeds = mbed_lstools.create()
+>>> mbeds.list_mbeds(fs_interaction=FSInteraction.BeforeFilter,
+                                    filter_function=None,
+                                    unique_names=False,
+                                    read_details_txt=False)
+[{'target_id_mbed_htm': u'0240000032044e4500257009997b00386781000097969900', 'mount_point': 'D:', 'target_id': u'0240000032044e4500257009997b00386781000097969900', 'serial_port': u'COM18', 'target_id_usb_id': u'0240000032044e4500257009997b00386781000097969900', 'platform_name': u'K64F'}]
 ```
 
-***Note:*** ```mbed-ls``` tools pair only serial ports and mount points (not CMSIS-DAP - yet).
+#### Arguments
 
-We can see that on our host machine (running Ubuntu) there are many 'disk type' devices visible under ```/dev/disk```. The mbed boards can be distinguished and filtered by their unique ```USB-ID``` conventions. In our case, we can see pairs of ```usb-ids``` in both ```/dev/serial/usb-id``` and ```/dev/disk/usb-id``` with embedded ``` TargetID```.  ```TargetID``` can be filtered out, for example using this sudo-regexpr: ```(“MBED”|”mbed”|”STMicro”)_([a-zA-z_-]+)_([a-zA-Z0-9]){4,}```
+##### `fs_interaction`
 
-For example, we can match the board 066EFF525257775087141721 by connecting a few dots:
+**Default:** `FSInteraction.BeforeFilter`
 
-* ```usb-MBED_microcontroller_066EFF525257775087141721-0:0 -> ../../sdd```
-* ```usb-STMicroelectronics_STM32_STLink_066EFF525257775087141721-if02 -> ../../ttyACM2``` Based on the TargetID hash.
+This argument controls the accuracy and speed of this function. There are three choices (in ascending order of accuracy and decreasing order of speed):
 
-From this we know that the target platform has these properties:
+- `FSInteraction.NEVER` - This is the fastest option but also potentially the least accurate. It never touches the file system of the devices and uses only the information available through the OS. This is appropriate for use in a highly controlled environment (such as an automated Continuous Integration setup). **This has the potential to provide incorrect names and data. It may also lead to devices not being detected at all.**
```

Comment:
Nit: the filesystem is available through the OS. Could you rewrite "OS" to "USB descriptors"?